### ISSUE TITLE
Support for OwnCloud 10.x

### ISSUE THREAD

**Issue by adlerweb** [owncloud/tasks#382]
### Steps to reproduce
1. Install OwnCloud 10.x
2. Try to install Tasks

### Expected behaviour
Find Tasks in the new market

### Actual behaviour
1. Notice it is not available yet
2. Copy files manually
3. Patch files to get it recognized

With 10.x now distributed as stable the tasks app should aim to be compatible and possibly be integrated into the new market. At the moment _appinfo/info.xml_ limits installation to OC 9.2, changing to 10.0 and manually installing seems to work fine for regular tasks (WebUI add/remove/edit and DAV-sync via CalDAV-Sync on android using multiple lists).

- [ ] Do more checks
- [x] Bump info.xml for 10.x
- [ ] Check if additional modifications are needed to apply for a market listing
- [ ] Apply for marketplace.owncloud.com

**Comment by DeepDiver1975** [comment]
@adlerweb are you interested in taking care of this? Conributions are always welcome.

**Comment by el3ctrician** [comment]
i can modify the version and update the signeture if that all it takes .. can you confirm ? i am intrested in such a feature

**Comment by DeepDiver1975** [comment]
> ? i am intrested in such a feature

go for it - let me know if you need any support!

**Comment by el3ctrician** [comment]
i am sorry but i am a litelle bit confused ... should i sign the APP with my keys or should the original arthour use his keys to sign it ?

**Comment by el3ctrician** [comment]
no share option n lists, is this a normal behavioural ?
![screenshot from 2017-05-11 14-09-38](https://cloud.githubusercontent.com/assets/17578910/25948405/93520390-3653-11e7-9422-e522d3b458bf.png)

**Comment by DeepDiver1975** [comment]
> i am sorry but i am a litelle bit confused ... should i sign the APP with my keys or should the original arthour use his keys to sign it ?
> after reading the documentation i think the owner should sign it .. i was not able to find the public key even

keys are per app basis - @raimund-schluessler has the key - maybe he can prepare an app package ... if not we can revoke the key and @el3ctrician you are free to generate your own.

**Comment by el3ctrician** [comment]
if @raimund-schluessler is not avaliable i can try to prepare a package and push it to the market.. just let me know

**Comment by raimund-schluessler** [comment]
Sorry guys, I moved to N*cloud and neither have an ownCloud installation to test and sign it at the moment, nor do I have an account at the new ownCloud appstore. Also, I am on vacation for the next two weeks. Therefore, I think the best (and most future proof) solution is to revoke the certificate and let @el3ctrician prepare the package for the appstore. If you need any information let me know. Thanks.

**Comment by lemmy04** [comment]
I have installed a git snapshot which works fine but gives me integrity warnings because it is unsigned. PLEASE someone prepare an official release pack of this.

**Comment by Lerentis** [comment]
same as @lemmy04 . the only thing i see is missing to support owncloud 10 is a signature and a release. Any ETA on this @el3ctrician || @raimund-schluessler ?

**Comment by raimund-schluessler** [comment]
@Lerentis I am sorry, but I do not maintain this app anymore. I now work on https://github.com/nextcloud/tasks

**Comment by sarahbesic** [comment]
@Lerentis What’s needed to do this tasks (signature + release)? I’d do it for Gods sake cause I need this app w/ OC 10.0.2.1

**Comment by raimund-schluessler** [comment]
How signing an app works is described here: https://doc.owncloud.org/server/10.0/developer_manual/app/code_signing.html

But you can also just put the app in the apps folder of your owncloud instance and live with the signature warnings. Not nice, but it works.

Everything else needs help from the owncloud guys.

**Comment by Lerentis** [comment]
@sarahbesic or you could disable the consistency check in your config.php
```
'integrity.check.disabled' => true,
```
not recommended, but it works

**Comment by linkp** [comment]
I don't want to put words in anyone's mouth, but I think what is being asked here is how do we get the ownCloud guys to assign this to a new maintainer who is able to get this package signed and distributed through the marketplace, so we can clear out the integrity warning without resorting to disabling the check.

**Comment by Sh4d0wF1gh7** [comment]
Hello, you can use the signature.json directly from a Nextcloud -> Tasks App...
Perhaps a googd idea to copy this file here in the appinfo folder as new compatible version with owncloud 10.x? Also the info.xml should be updated with new MAX Owncloud Version.

**Comment by sarahbesic** [comment]
@raimund-schluessler @Lerentis : Thx 4 your support, that’s what I did so long, but @linkp got what I meant. Everything else might be at best a wobbly workaround.
@Sh4d0wF1gh7 Could it be found here: https://github.com/nextcloud/tasks ?

**Comment by Sh4d0wF1gh7** [comment]
Sorry my last post don´t solve the problem.
I inserted the signature.json from a NextCloud Task Installation but the error:
`App Certificate is not valid.`
appears also with this.
Someone have to do to create a certificate.
The Tasks App is running in OwnCloud 10.0.3 but with this Certificate error.

**Comment by dlandon** [comment]
Is anyone going to take over this app and get it published in the marketplace?  If there are no takers, I could see what I can do.

**Comment by Sh4d0wF1gh7** [comment]
Would be great if someone can do this.

**Comment by dlandon** [comment]
Who do I talk to about access to the github?  And how to get the app into the market?

**Comment by DeepDiver1975** [comment]
> How signing an app works is described here: https://doc.owncloud.org/server/10.0/developer_manual/app/code_signing.html

@dlandon this is the doc link about signing - if you are interested I can add you to the github repo here. I'll revoke the key of @raimund-schluessler as of now ...

**Comment by dlandon** [comment]
Yes, please add me to the github repo.  Your link to the code signing doc doesn't look.  I'll try to find it on my own.

**Comment by DeepDiver1975** [comment]
ooops - https://doc.owncloud.org/server/10.0/developer_manual/app/advanced/code_signing.html?highlight=signing#how-to-get-your-app-signed

**Comment by DeepDiver1975** [comment]
@dlandon invitation jst sent out - welcome to the show!

**Comment by dlandon** [comment]
Invitation accepted.

I'm working on the signature file.  Is there a script I can use to generate the hashes in the proper format for the signature file?

**Comment by DeepDiver1975** [comment]
@dlandon try running the makefile - https://github.com/owncloud/tasks/blob/master/Makefile#L14

```bash
make sign
```
 
should do what ever is necessary ...

**Comment by dlandon** [comment]
I can't figure out how to post the CSR on https://github.com/owncloud/appstore-issues.  Nothing is accepted.

**Comment by dlandon** [comment]
I have the app signed and ready to go.  I will upload to github later today after I do some more checking.  I will then work on getting it on the market.  Once that is done, I'll work on updating the app to match the work done on the Nextcloud app if appropriate.  I don't know if there are any changes specific to Nextcloud.

**Comment by Sh4d0wF1gh7** [comment]
PERFECT...
Thank you so much. Looking forward for the new Task App for my loved ownCloud.

**Comment by dlandon** [comment]
I'm unable to push my changes because I don't have write access.  I guess someone has to approve the push.  I'm used to owning and having control over my github repositories.  This is different for me.

**Comment by DeepDiver1975** [comment]
@dlandon the master branch is protected against direct pushes. You have to work on branches and open pull requests. Let me know if you need further assistance.

**Comment by dlandon** [comment]
It looks like I've done the pull request and added my changes.  Where can I find documentation on how to publish an app in the marketplace?  When I try to upload my tasks.tar.gz file I get a signature file not found issue.  I'm not sure what it wants.

**Comment by DeepDiver1975** [comment]
> When I try to upload my tasks.tar.gz file I get a signature file not found issue.

either the file is missing in the tar ball or the folder layout is not as expected and the file could not be found.

**Comment by dlandon** [comment]
The signature.json file?

**Comment by DeepDiver1975** [comment]
> The signature.json file?

yes

**Comment by dlandon** [comment]
Version 0.9.3 has been uploaded to the Market.
